### ISSUE TITLE
[SURE-4241] Reduce network traffic between cattle-cluster agent to/from Rancher on nodes with limited network connectivity

### ISSUE THREAD

**Issue by gaktive** [rancher/rancher#41908]

**Comment by gaktive** [comment]
Internal reference: SURE-4241
Reported in 2.6.3

Request description:
Rancher cattle cluster agents need a limited connectivity mode for far edge environments.

The configuration of the same should be exposed to the end users to configure how frequently they would like the cluster-agent / rancher to reconcile the state of the downstream cluster.

One user has done some bench marking and noticed that pre-2.6.3, a single node k3s cluster would generate around 190MB of data during a 24 hour window, even when no actual actions were being performed on the said downstream cluster. In 2.6.3, this has reduced significantly to around 50MB/day. However, we should look into reducing this further. There is one specific large scale use case involving upwards of 6000 mobile endpoints where the current chattiness makes it costly.

Proposal: an additional tuning option in the cattle cluster agent so that the Rancher check-in can be customized for less regular polls or a logic to detect if an endpoint has limited connectivity and reduce comms.

**Comment by strowi** [comment]
Yes please! We are currently starting to deploy some test-devices with LTE and noticed only fleet-agent + cattle-cluster-agen each send ~ 2,9kb/s on average. Any tips on how to reduce this further with the current situation (rancher-2.7.4)?

**Comment by ApolloDS** [comment]
add a comment so the label can be removed.

**Comment by moio** [comment]
still relevant

**Comment by moio** [comment]
still relevant

**Comment by femtonelson** [comment]
Up to 20MB of data per hour consumed by the Rancher server connection (v2.7.9). 99% of this data is consumed in uplink direction!

**Comment by ApolloDS** [comment]
still relevant

**Comment by moio** [comment]
still relevant

**Comment by moio** [comment]
this is still being worked on

**Comment by moio** [comment]
still relevant
